atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/CombinedLocksCondition.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.palantir.lock.HeldLocksToken;
+import com.palantir.lock.LockService;
+
+public class CombinedLocksCondition implements AdvisoryLocksCondition {
```

Comment:
We're splitting the conditions up so as you mention we now make two separate calls to timelock for requests with advisory locks. Probably fine though, especially since the external condition shortcircuits if empty.